codegen/test/rtc/src/tmp_dir.cpp

```diff
@@ -31,18 +35,56 @@ std::string unique_string(const std::string& prefix)
 }
 
 tmp_dir::tmp_dir(const std::string& prefix)
-    : path(std::filesystem::temp_directory_path() /
-           unique_string(prefix.empty() ? "ck-rtc" : "ck-rtc-" + prefix))
+    : path(get_tmp_dir_path() + "/" + unique_string(prefix.empty() ? "ck-rtc" : "ck-rtc-" + prefix))
 {
-    std::filesystem::create_directories(this->path);
+    new_dir(this->path);
 }
 
+std::string tmp_dir::get_tmp_dir_path()
+{
+    // use getenv to get the path of the tmp dir
+    const char* tmp_dir_path = std::getenv("TMPDIR");
+
+    if(tmp_dir_path == nullptr)
+    {
+        return "/tmp";
+    }
+    return tmp_dir_path;
+}
+
+// TODO: finish this method
+void tmp_dir::new_dir(std::string dir_path)
+{
+    int created = mkdir(dir_path.c_str(), 0755);
+    if(created != 0)
+    {
+        throw std::runtime_error("Directory was not created");
+    }
+}
+
+bool tmp_dir::exists(std::string check_path)
+{
+    struct stat sb;
+    if(stat(check_path.c_str(), &sb) == 0 && !(sb.st_mode & S_IFDIR))
```

Comment:
Could you please use bitwise `AND-ing` directly with `sb.st_mode`:  like `(sb.st_mode & S_IFMT) == S_IFDIR; `  ~ to extract the file type code, and compared to the appropriate constant.